src/stirling/source_connectors/socket_tracer/protocols/mongodb/types.h

```diff
@@ -153,10 +155,21 @@ struct Record {
   }
 };
 
+using stream_id_t = int32_t;
+struct State {
+  std::vector<std::pair<mongodb::stream_id_t, bool>> stream_order;
```

Comment:
Now that I better understand the PR, I'm wondering if a comment here will be useful. For a while, I could not figure out why there was a bool in this, nor what it even did. I'm thinking the comment would look like:
```
// The stream_order state tracks which stream_id to stitch first.
// In more detail, the mongo-db wire protocol can link responses together having each response
// point to a new response via the stream-id. To stitch correctly, we record the first stream-id in each
// such sequence during frame parsing, and store that in the stream_order vector.
// The second item in the pair is a boolean flag that is initially false, but set to true when the entry
// becomes safe to remove.
```

Reply:
Added, thanks for this